### ISSUE TITLE
关于user和item特征

### ISSUE THREAD

**Issue by ryy980622** [JarenceSJ/ReviewGraph#3]
您好，我在阅读代码rgc.py的时候发现user和item特征定义是在GCMCGraphConv模块里的self.weight，这样做是不是对每个rating对应的图都初始化了一组user和item特征？以Digital_Music为例，代码共定义了2*5组user和item的特征，然而论文中eq.(1)中写的是所有rating的图上传播的是共享的一组user和item特征，请问这个地方是论文中写错了吗？

**Comment by FKCHAN** [comment]
您好，我在调试**bert_whitening.py程序**时遇到了一个问题，
运行word2vec.py和load_data.py都没有问题，
但是，调试**bert_whitening.py**时，

执行输出是：
pre tokenize: 100%|██████████| 64706/64706 [02:42<00:00, 399.34it/s]
  0%|          | 0/506 [00:00<?, ?it/s]

会报错：
ValueErrors: too many values to unpack (excepted 2)
报错line90： 
def collate_fn(data):
    u, i, r, tokens = zip(*data)
    **encoding = bert_tokenizer(tokens, return_tensors='pt', padding=True,
                              truncation=True, is_pretokenized=True)**

    return torch.Tensor(u), torch.Tensor(i), torch.Tensor(r), \
        encoding['input_ids'], encoding['attention_mask']

请问能帮我解答一下么？或者提供除amazon数据集以外完整的文件和环境版本，不胜感激
frankchan92@163.com

**Comment by JarenceSJ** [comment]
> 您好，我在阅读代码rgc.py的时候发现user和item特征定义是在GCMCGraphConv模块里的self.weight，这样做是不是对每个rating对应的图都初始化了一组user和item特征？以Digital_Music为例，代码共定义了2*5组user和item的特征，然而论文中eq.(1)中写的是所有rating的图上传播的是共享的一组user和item特征，请问这个地方是论文中写错了吗？

因为每个 rating 的 graph 都对 embeding 做了映射，参考 DGL 实现的 GC-MC，就直接分开对五个 rating 子图分别定义了 embedding。

**Comment by JarenceSJ** [comment]
> 您好，我在调试**bert_whitening.py程序**时遇到了一个问题， 运行word2vec.py和load_data.py都没有问题， 但是，调试**bert_whitening.py**时，
> 
> 执行输出是： pre tokenize: 100%|██████████| 64706/64706 [02:42<00:00, 399.34it/s] 0%| | 0/506 [00:00<?, ?it/s]
> 
> 会报错： ValueErrors: too many values to unpack (excepted 2) 报错line90： def collate_fn(data): u, i, r, tokens = zip(*data) **encoding = bert_tokenizer(tokens, return_tensors='pt', padding=True, truncation=True, is_pretokenized=True)**
> 
> ```
> return torch.Tensor(u), torch.Tensor(i), torch.Tensor(r), \
>     encoding['input_ids'], encoding['attention_mask']
> ```
> 
> 请问能帮我解答一下么？或者提供除amazon数据集以外完整的文件和环境版本，不胜感激 [frankchan92@163.com](mailto:frankchan92@163.com)

环境和数据参考这个 [issue](https://github.com/JarenceSJ/ReviewGraph/issues/2#issuecomment-1140393149)。
感谢提醒，我更新一下 readme。
